/*
 * Copyright (C) 2025 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package android.hardware.automotive.vehicle.proto;

import "android/hardware/automotive/vehicle/PropIdAreaId.proto";
import "android/hardware/automotive/vehicle/RawPropValues.proto";
import "android/hardware/automotive/vehicle/StatusCode.proto";

message SupportedValuesListResult {
    /**
     * The status for result. If this is not OK, the operation failed for this
     * [propId, areaId].
     */
    StatusCode status = 1;
    /**
     * The supported values list.
     *
     * If the [propId, areaId] does not specify a supported values list, this
     * is {@code null}.
     *
     * This must be ignored if status is not {@code StatusCode.OK}.
     */
    repeated RawPropValues supported_values_list = 2;
};

message GetSupportedValuesListsRequest {
    repeated PropIdAreaId prop_id_area_id = 1;
};

message GetSupportedValuesListsResult {
    repeated SupportedValuesListResult result = 1;
};
